### PR TITLE
RavenDB-13187 - fix backup location destination folders

### PR DESCRIPTION
**Pull request by Or-Geva**

### CODE REVIEW

**Comment by Or-Geva** [comment]
https://issues.hibernatingrhinos.com/issue/RavenDB-13187

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Web/Studio/FolderPath.cs [hunk 1]
```diff
@@ -20,7 +23,25 @@ public static FolderPathOptions GetOptions(string path, bool isBackupFolder, Rav
                 }
                 else if (string.IsNullOrWhiteSpace(path))
                 {
-                    folderPathOptions.List.AddRange(GetAvailableDrives());
+                    var availableDrivcs = GetAvailableDrives();
```
Comment (ppekrol) [inline src/Raven.Server/Web/Studio/FolderPath.cs]: typo

File: src/Raven.Server/Web/Studio/FolderPath.cs [hunk 2]
```diff
@@ -110,6 +136,11 @@ private static string GetRestrictedFolder(bool isBackupFolder, RavenConfiguratio
 
     public class FolderPathOptions
     {
-        public List<string> List { get; } = new List<string>();
+        public List<string> List { get; private set; } = new List<string>();
```
Comment (ppekrol) [inline src/Raven.Server/Web/Studio/FolderPath.cs]: Maybe it should be sorted list, then we will not have to sort it later on?

File: src/Raven.Server/Web/Studio/FolderPath.cs [hunk 3]
```diff
@@ -110,6 +136,11 @@ private static string GetRestrictedFolder(bool isBackupFolder, RavenConfiguratio
...
+
+        public void SortedList()
```
Comment (ppekrol) [inline src/Raven.Server/Web/Studio/FolderPath.cs]: method name should be `Sort`

File: src/Raven.Server/Web/Studio/FolderPath.cs (hunk 2 above)
Comment (Or-Geva) [reply src/Raven.Server/Web/Studio/FolderPath.cs]: In windows, it is already sorted.

File: src/Raven.Server/Web/Studio/FolderPath.cs (hunk 1 above)
Comment (Or-Geva) [reply src/Raven.Server/Web/Studio/FolderPath.cs]: Fixed

File: src/Raven.Server/Web/Studio/FolderPath.cs (hunk 3 above)
Comment (Or-Geva) [reply src/Raven.Server/Web/Studio/FolderPath.cs]: Fixed

File: src/Raven.Server/Web/Studio/FolderPath.cs [hunk 4]
```diff
@@ -64,6 +85,11 @@ public static FolderPathOptions GetOptions(string path, bool isBackupFolder, Rav
                         }
                     }
                 }
+
+                 if (PlatformDetails.RunningOnPosix)
+                {
```
Comment (ppekrol) [inline src/Raven.Server/Web/Studio/FolderPath.cs]: please follow formatting rules

File: src/Raven.Server/Web/Studio/FolderPath.cs [hunk 5]
```diff
@@ -110,6 +136,11 @@ private static string GetRestrictedFolder(bool isBackupFolder, RavenConfiguratio
 
     public class FolderPathOptions
     {
-        public List<string> List { get; } = new List<string>();
+        public List<string> List { get; private set; } = new List<string>();
+
+        public void Sort()
+        {
+            List = List.OrderBy( O => O ).ToList();
```
Comment (ppekrol) [inline src/Raven.Server/Web/Studio/FolderPath.cs]: Please follow formatting rules. Also it should be `(o => o)`

File: src/Raven.Server/Web/Studio/FolderPath.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Server/Web/Studio/FolderPath.cs]: Do we care that much? It is not a hot-path? How many items do we have there on average? 20?

Comment (Or-Geva) [reply src/Raven.Server/Web/Studio/FolderPath.cs]: Ok, Changed to `SortedSet`.
SortedList uses 2 generic types

File: src/Raven.Server/Web/Studio/FolderPath.cs (hunk 5 above)
Comment (Or-Geva) [reply src/Raven.Server/Web/Studio/FolderPath.cs]: Removed

File: src/Raven.Server/Web/Studio/FolderPath.cs (hunk 4 above)
Comment (Or-Geva) [reply src/Raven.Server/Web/Studio/FolderPath.cs]: Done.

**Pull request merged**